tools/clang/lib/Sema/SemaHLSL.cpp

```diff
@@ -4665,9 +4670,10 @@ static bool CombineObjectTypes(ArBasicKind Target, _In_ ArBasicKind Source,
       return true;
     }
     break;
+  default:
+    // Not a combinable target.
+    AssignOpt(AR_BASIC_UNKNOWN, pCombined);
   }
-
-  AssignOpt(AR_BASIC_UNKNOWN, pCombined);
```

Comment:
I think in some cases (where the case matches, but the if condition inside the case isn't true), AssignOpt used to be executed, but now won't.

Reply:
Yeah. it looks like you're right. I'll move it back out and make it a break.